canonical-web-and-design/jaas-dashboard
# Create configuration panel UI

## Done

Create the UI for fetching and displaying the configuration values.

Note: There are a few features yet to be implemented, see the associated Epic.

## QA

- Pull code
- Run `./run clean && ./run serve`
- Open http://0.0.0.0:8036/
- Load the `charmed-kubernetes` model and click on `containerd`. This is a good config because it has both a string and boolean type.
- [ ] When viewing the configuration pane it should show a spinner while it's loading.
- [ ] It should render both textarea and boolean config UI's
- [ ] When changing a configuration value a `use default` UI element should appear and a `reset all fields` button should appear at the top
- [ ] When focusing on an input or clicking around a config option it should show the description on the right and highlight the input.
- [ ] Clicking the `use default` button should change the field to the default 
- [ ] Clicking the `reset all` should reset all the fields to the default.
- [ ] After modifying an input and saving, it should actually save (you can check this by closing and re-opening the panel)
- [ ] Play around with modifying and restoring values, to see if it all works as expected.
- [ ] Hitting cancel should close the panel.

## Details

Fixes https://github.com/canonical-web-and-design/juju-squad/issues/1567
Fixes https://github.com/canonical-web-and-design/juju-squad/issues/1571
Fixes https://github.com/canonical-web-and-design/juju-squad/issues/1566
Fixes https://github.com/canonical-web-and-design/juju-squad/issues/1569
Fixes https://github.com/canonical-web-and-design/juju-squad/issues/1568
Fixes https://github.com/canonical-web-and-design/juju-squad/issues/1573
Fixes https://github.com/canonical-web-and-design/juju-squad/issues/1583
Fixes https://github.com/canonical-web-and-design/juju-squad/issues/1572
Fixes https://github.com/canonical-web-and-design/juju-squad/issues/1570

## Screenshots

<img width="1680" alt="Screen Shot 2021-01-12 at 3 38 29 PM" src="https://user-images.githubusercontent.com/532033/104377593-5f21d580-54ec-11eb-88cd-b249f9ca9e11.png">
